foxai/explainer/computer_vision/algorithm/input_x_gradient.py

```diff
@@ -20,194 +22,150 @@ class BaseInputXGradientSHAPCVExplainer(Explainer):
-                argument of a Tensor or arbitrary (non-tuple) type or a tuple
+                can be provided. It must be a tuple
                 containing multiple additional arguments including tensors
                 or any arbitrary python types. These arguments are provided to
                 forward_func in order following the arguments in inputs.
                 Note that attributions are not computed with respect
                 to these arguments.
                 Default: None
-            attribute_to_layer_input: Indicates whether to
-                compute the attribution with respect to the layer input
-                or output. If `attribute_to_layer_input` is set to True
-                then the attributions will be computed with respect to
-                layer input, otherwise it will be computed with respect
-                to layer output.
-                Default: False
-            layer: Layer for which attributions are computed.
-                If None provided, last convolutional layer from the model
-                is taken.
-                Default: None
 
         Returns:
             The input x gradient with respect to each input feature or gradient
             and activation for each neuron in given layer output. Attributions
-            will always be the same size as the provided inputs, with each value
-            providing the attribution of the corresponding input index.
-            If a single tensor is provided as inputs, a single tensor is
-            returned. If a tuple is provided for inputs, a tuple of
-            corresponding sized tensors is returned.
+            will always be the same size as the provided input.
 
         Raises:
             RuntimeError: if attribution has shape (0).
         """
-        attributions: AttributionsType
-        input_x_gradient = self.create_explainer(model=model, layer=layer)
-
-        if isinstance(input_x_gradient, LayerGradientXActivation):
-            attributions = input_x_gradient.attribute(
-                input_data,
-                target=pred_label_idx,
-                additional_forward_args=additional_forward_args,
-                attribute_to_layer_input=attribute_to_layer_input,
-            )
-        else:
-            attributions = input_x_gradient.attribute(
-                input_data,
-                target=pred_label_idx,
-                additional_forward_args=additional_forward_args,
-            )
-        validate_result(attributions=attributions)
-        return attributions
 
+        grad_required = input_data.requires_grad
+        input_data.requires_grad_()
 
-class InputXGradientCVExplainer(BaseInputXGradientSHAPCVExplainer):
-    """Input X Gradient algorithm explainer."""
-
-    def create_explainer(
-        self,
-        model: ModelType,
-        **kwargs,
-    ) -> Union[InputXGradient, LayerGradientXActivation]:
-        """Create explainer object.
+        gradient = compute_gradients(
```

Comment:
I am wondering if it is necessary to clear gradients before this function, i.e. by calling `model.zero_grad()`. Have you checked that?

Reply:
It wasn't in the previous implementation.

Reply:
Could you check if this has no effect on the resulting gradients when computing gradients multiple times in a row on the same model with and without `model.zero_grad()` calls?

Reply:
I checked and gradient results are repeatable without or with model.zero_grad().